### ISSUE TITLE
Different differentials file when reproducing atopic dermatitis set

### ISSUE THREAD

**Issue by L-Boonstra** [knightlab-analyses/reference-frames#10]
Hello,

I’m trying to reproduce the results of the atopic dermatitis dataset.
In Linux I used the following code to create the environment:
```
conda create -n songbird_env numpy=1.15.4 scikit-bio=0.5.5 seaborn pandas=0.23.4 python=3.6 -c conda-forge
source activate songbird_env
conda install tensorflow=1.10 tqdm nomkl
conda install biom-format h5py -c conda-forge
conda install jupyter notebook
conda install songbird -c conda-forge
```

I had to add python=3.6 to the first line for tensorflow and songbird to install.
Then I used the code from byrd_multinomial.sh to get the differentials file
```
songbird multinomial \
	  --input-biom ../data/byrd_skin_table.biom \
	  --metadata-file ../data/byrd_metadata.txt \
	  --formula "C(Timepoint, Treatment('F'))" \
	  --training-column Test \
	  --epochs 5000 \
	  --learning-rate 1e-3 \
	  --batch-size 5 \
	  --differential-prior 10 \
	  --summary-dir ../results/byrd-results
```

The differential file that is in the result folder in the repository has 2860 rows, but the file that I got has only 1918 rows. The file below is the one that I got.

[differentials_V3.txt](https://github.com/knightlab-analyses/reference-frames/files/4040452/differentials_V3.txt)

Some more information:

If I check the fit I get the following graphs:

![Capture](https://user-images.githubusercontent.com/22447867/72069836-5b131480-32e8-11ea-9594-6baf09ea4189.PNG)

I also tried out the code from figure3.ipynb till the first graph with the differentials file that I got:

![byrd-post-flare-ranks](https://user-images.githubusercontent.com/22447867/72069911-84cc3b80-32e8-11ea-920a-1a549edd48cd.png)

So I'm wondering why my differentials file is different then the one in the results folder.

Thank you in advance for the help.

**Comment by mortonjt** [comment]
Ahhh.   I bet you that the discrepancy that you are seeing is due to a change in the songbird defaults.

In the paper that we published, we dropped taxa that appeared in less than 5 samples.
But in a newer release of songbird, we dropped taxa that appeared in less than 10 samples.

Below are the lines of code to show this difference
[before](https://github.com/biocore/songbird/blob/e7962829c7188a8e17ecc963a62cfef28e2885c9/scripts/songbird#L50)
[after](https://github.com/biocore/songbird/blob/master/songbird/parameter_info.py#L63)

If you want to reproduce the *exact* results in the paper, you need to use songbird version 0.8.2

If you are using pip, this can be done with

```
pip install songbird==0.8.2
```

With [conda](https://anaconda.org/conda-forge/songbird), this is
```
conda install songbird=0.8.2 -c conda-forge
```

Let me know if you can reproduce the results with the older version -- then I can update the readme.  Thank you for reporting.

**Comment by L-Boonstra** [comment]
I've tried it again with songbird=0.8.2 and now I'm getting the following file with 2125 rows:

[differentials_OV.txt](https://github.com/knightlab-analyses/reference-frames/files/4044790/differentials_OV.txt)

So I have more taxa showing up in my differentials file now, but I still don't get the same results.

**Comment by mortonjt** [comment]
Hi @LieseBoonstra , the feature miscounting is due to the `min-feature-count` option, that should be set to `min-feature-count=0`.  So the command will look like as follows

```
songbird multinomial \
	  --input-biom ../data/byrd_skin_table.biom \
	  --metadata-file ../data/byrd_metadata.txt \
	  --formula "C(Timepoint, Treatment('F'))" \
	  --training-column Test \
	  --epochs 5000 \
	  --learning-rate 1e-3 \
	  --batch-size 5 \
          --min-feature-count 0 \
	  --differential-prior 10 \
	  --summary-dir ../results/byrd-results
```

Note that the results aren't expected to be exactly the same due to the initial random seed.  But this should resolve the filtering inconsistency.

**Comment by L-Boonstra** [comment]
I got all the rows now, thank you!

**Comment by mortonjt** [comment]
Awesome! Glad to hear that this has been resolved
